Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/TextInput/TextInputViewController.swift

```diff
@@ -0,0 +1,139 @@
+            barButton(keyboardDoneButton)
+        }
+    }
+    
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        if viewModel.rightBarButtonTitle != nil {
+            self.navigationItem.rightBarButtonItem = .init(title: viewModel.rightBarButtonTitle?.value,
+                                                           style: .done,
+                                                           target: self,
+                                                           action: #selector(dismissScreen))
+        }
+    }
+
+    public func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+
+        return viewModel.textFieldViewModel.validator(existingString: textField.text, range: range, replacementString: string)
+    }
+    
+    public func textFieldDidEndEditing(_ textField: UITextField) {
+        if let text = textField.text {
+            if !text.isEmpty && Double(text) != nil {
+                primaryButton.isEnabled = true
+            } else {
+                textField.text = ""
+                primaryButton.isEnabled = false
+            }
+        } else {
+            textField.text = ""
+            primaryButton.isEnabled = false
+        }
+    }
+    
+    @IBOutlet private var scrollView: UIScrollView!
+    
+    @IBOutlet private var titleLabel: UILabel! {
+        didSet {
+            titleLabel.text = viewModel.title.value
+            titleLabel.font = .largeTitleBold
+            titleLabel.accessibilityTraits = .header
+            titleLabel.accessibilityIdentifier = "titleLabel"
+        }
+    }
+    
+    @IBOutlet private var innerStackView: UIStackView!
+    
+    @IBOutlet private var textField: UITextField! {
+        didSet {
+            textField.keyboardType = viewModel.textFieldViewModel.keyboardType
+            textField.font = .body
+            textField.placeholder = viewModel.textFieldViewModel.placeholder
+            textField.accessibilityIdentifier = "textField"
+        }
+    }
+    
+    @IBOutlet private var textFieldFooter: UILabel! {
+        didSet {
+            textFieldFooter.text = viewModel.textFieldFooter
+            textFieldFooter.font = .footnote
+            textFieldFooter.textColor = .secondaryLabel
+            textFieldFooter.numberOfLines = 0
+            textFieldFooter.accessibilityIdentifier = "textFieldFooter"
+            textFieldFooter.isHidden = viewModel.textFieldFooter == nil
+        }
+    }
+    
+    @IBOutlet private var primaryButton: RoundedButton! {
+        didSet {
+            primaryButton.setTitle(viewModel.buttonViewModel.title, for: .normal)
+            primaryButton.accessibilityIdentifier = "primaryButton"
+        }
+    }
+    
+    
+    @IBAction private func primaryButtonAction(_ sender: Any) {
+        guard let text = textField.text else {
+            // failed to get text from text field
```

Comment:
would we want an error thrown here or anything to happen when this is hit?

Reply:
I don't think so, but perhaps one for a wider discussion. A text field should always have a `text` or `nil`. If it is `nil` then I think the correct behaviour is to do nothing.

Reply:
I suppose its maybe one for design, if the text field is nil when the user presses the primary action button, you'd think that you would get some feedback to say why the button is not doing anything.

Reply:
On the screen, the button is disabled until editing has finished (ie tapping the `done` button on the keyboard toolbar) and the textfield is not empty.

Perhaps I could add some logic here to set the primary button to disabled if there's no text in the text field as a belt and braces approach.